plugins/performance-lab/includes/server-timing/class-perflab-server-timing.php

```diff
@@ -288,19 +288,43 @@ function ( string $output, ?int $phase ): string {
 	 */
 	private function format_metric_header_value( Perflab_Server_Timing_Metric $metric ): ?string {
 		$value = $metric->get_value();
-
-		// If no value is set, make sure it's just passed through.
-		if ( null === $value ) {
-			return null;
-		}
+		$start = $metric->get_start_time();
+		$desc  = $metric->get_description();
 
 		if ( is_float( $value ) ) {
 			$value = round( $value, 2 );
 		}
 
+		if ( is_float( $start ) ) {
+			$start = round( $start, 2 ) * 1000.00;
+		}
+
+		// If no value is set, make sure it's just passed through.
+		if ( null === $start && null === $value ) {
+			return null;
+		}
+
 		// See https://github.com/WordPress/performance/issues/955.
 		$name = preg_replace( '/[^!#$%&\'*+\-.^_`|~0-9a-zA-Z]/', '-', $metric->get_slug() );
 
-		return sprintf( 'wp-%1$s;dur=%2$s', $name, $value );
+		if ( 'response-start' !== $name && 'response-end' !== $name ) {
+			$name = 'wp-' . $name;
+		}
+
+		$header_value = sprintf( '%s', $name );
+
+		if ( null !== $value ) {
+			$header_value .= sprintf( ';dur=%s', $value );
```

Comment:
Technically shouldn't this be formatted as a float?

```suggestion
			$header_value .= sprintf( ';dur=%f', $value );
```

Maybe not, because for some reason this seems to result in a loss of precision! https://3v4l.org/oOVJX

I consulted with Gemini: https://g.co/gemini/share/7e25d0065550

Looks like if you want to retain the precision then you have to explicitly pass the desired precision, like `%.10f` but then you can get a lot of extra zeros when you don't need them!

So nevermind :)